OpenEdXMobile/src/main/java/org/edx/mobile/model/api/EnrollmentResponse.kt

```diff
@@ -64,13 +65,30 @@ data class EnrollmentResponse(
 }
 
 /**
- * Method to filter the audit course Skus from the given enrolled course list.
+ * Method to filter the audit courses from the given enrolled course list.
  *
- * @return the list of all audit courses SKUs.
+ * @return the list of all audit courses with non-null Skus.
  */
-fun List<EnrolledCoursesResponse>.getAuditCoursesSku(): List<String> {
-    return this
-        .filter { it.isAuditMode }
-        .mapNotNull { it.courseSku }
-        .toList()
+fun List<EnrolledCoursesResponse>.getAuditCourses(): List<AuditCourse> {
+    return this.filter {
+        EnrollmentMode.AUDIT.toString().equals(it.mode, true) && it.courseSku.isNullOrBlank().not()
+    }.mapNotNull { course ->
+        course.courseSku?.let { sku ->
+            AuditCourse(
+                id = course.courseId,
+                sku = sku,
+                isSelfPaced = course.course.isSelfPaced
+            )
+        }
+    }.toList()
 }
+
+/**
+ * Custom model class to store audit courses data for unfulfilled purchases
+ */
+data class AuditCourse(
```

Comment:
IMO better to remove replace `AuditCourse` with `IAPFlowData` and rename it with  `IAPFlowData`->`AuditCourse`